cli/cmd/stat.go

```diff
@@ -310,6 +315,7 @@ func writeStatsToBuffer(rows []*pb.StatTable_PodGroup_Row, w *tabwriter.Writer,
 				tcpOpenConnections: r.GetTcpStats().GetOpenConnections(),
 				tcpReadBytes:       getByteRate(r.GetTcpStats().GetReadBytesTotal(), r.TimeWindow),
 				tcpWriteBytes:      getByteRate(r.GetTcpStats().GetWriteBytesTotal(), r.TimeWindow),
+				hasRequestData:     statHasRequestData(r.Stats),
```

Comment:
Would it be simpler to make this `if r.Stats != nil && statHasRequestData(r.Stats)` on line 308 instead of adding a new boolean field to this struct?  That way, `stats[key].rowStats` will be nil in the case where there is no traffic and the existing empty template code path will be used.

Reply:
Yes I did it like that initially, but then I realized that very similar logic needs to be applied to the [routes command](https://github.com/linkerd/linkerd2/pull/3649/files#diff-e87f0b2931093d08df1cea47d36797a6R204-R224). And there the difference is that there is no `nil` added to the list you are iterating over. So instead of forcefully adding a nil there or skipping the row altogether, which is not an option, I decided to unify the logic between the two places a bit and rely on the bool. I can change it to rely on a `nil` being present if you think its better. Your call!

Reply:
Ah, I see what you mean.  The routes row handling code seems either too coupled, or not coupled enough to the stat row code 😂 

Given that, I think my preference would be to handle this with a nil for stat and do something different for routes.  At the time when we are rendering the row, we can check the counts to see if there are no requests and render the blank line instead.  I don't think we need to persist that boolean in the struct.

Reply:
I edited it and took the bool approach to the routes part only. Hope that is better now